Helper/Api/Orders.php

```diff
@@ -934,4 +942,48 @@ private function getTaxRateId(Item $item, ?int $taxFreeId, array $rates): ?int
 
         return $taxRateId;
     }
+
+    /**
+     * @param int|float|string $price
+     * @return Money
+     */
+    private function getBrickMoneyPrice($price): Money
+    {
+        return Money::of($price, $this->order->getStoreCurrencyCode());
+    }
+
+    /**
+     * @param int|float|string $price
+     * @return float
+     */
+    private function getPriceValueForPayload($price): float
+    {
+        return $this->getBrickMoneyPrice($price)->getAmount()->toFloat();
+    }
+
+    /**
+     * @param Money $brickMoneyObj
+     * @return float
+     */
+    private function getPriceByBrickMoneyObj(Money $brickMoneyObj): float
+    {
+        return $brickMoneyObj->getAmount()->toFloat();
+    }
```

Comment:
```suggestion
    private function getPriceValueForPayload($price): float
    {
        return $this->getPriceByBrickMoneyObj($this->getBrickMoneyPrice($price));
    }

    /**
     * @param Money $brickMoneyObj
     * @return float
     */
    private function getPriceByBrickMoneyObj(Money $brickMoneyObj): float
    {
        return $brickMoneyObj->getAmount()->to($brickMoneyObj->getContext(), RoundingMode::HALF_UP);
    }
```

Perform rounding before getting the value